runtime/stdlib/account.go

```diff
@@ -658,21 +668,116 @@ func newAccountKeysGetFunction(
 					inter,
 					invocation.GetLocationRange,
 					accountKey,
-					// public keys are assumed to be already validated.
-					func(
-						_ *interpreter.Interpreter,
-						_ func() interpreter.LocationRange,
-						_ *interpreter.CompositeValue,
-					) error {
-						return nil
-					},
+					assumePublicKeyIsValid,
 				),
 			)
 		},
 		sema.AccountKeysTypeGetFunctionType,
 	)
 }
 
+func newAccountKeysForEachFunction(
+	gauge common.MemoryGauge,
+	provider AccountKeyProvider,
+	addressValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	address := addressValue.ToAddress()
+	// the PublicKey in `forEachKey(_ f: (PublicKey): Bool): Void`
+	callbackTypeParams := []sema.Type{sema.AccountKeyType}
+
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			fnValue, ok := invocation.Arguments[0].(interpreter.FunctionValue)
+
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			inter := invocation.Interpreter
+			getLocationRange := invocation.GetLocationRange
+
+			newSubInvocation := func(key interpreter.Value) interpreter.Invocation {
+				return interpreter.NewInvocation(
+					inter,
+					nil,
+					[]interpreter.Value{key},
+					callbackTypeParams,
+					nil,
+					getLocationRange,
+				)
+			}
+
+			liftKeyToValue := func(key *AccountKey) interpreter.Value {
+				return NewAccountKeyValue(
+					inter,
+					getLocationRange,
+					key,
+					assumePublicKeyIsValid,
+				)
+			}
+
+			var invocationErr error
+
+			err := provider.IterateKeys(address, func(key *AccountKey) bool {
```

Comment:
This can probably be implemented by using `GetAccountKey` on each of the indices from 0 to `count` rather than relying on the FVM to implement another function, we want to keep the runtime interface as minimal as possible.